### DISCUSSION TITLE
Installation on Restricted Systems

### DISCUSSION THREAD

**Discussion by adi8v** [cocotb/cocotb#2535] [Proposals]
My RHEL8 box is behind a proxy. Attempting to install using the --proxy flag resulted in this:

```
(py36-venv-rhel8) [axh@axh-dt ~]$ pip install --proxy=https://XXXX:8888 cocotb[bus]
Collecting cocotb[bus]
  Using cached https://files.pythonhosted.org/packages/e0/be/113b2722dfdf2a44aad58e2ec416ab0a0ae0346bddb0d2cf2e596fcb2a60/cocotb-1.5.1.tar.                                                                   gz
    Complete output from command python setup.py egg_info:
    Download error on https://pypi.org/simple/setuptools_scm/: [Errno 101] Network is unreachable -- Some packages may not be found!
    Download error on https://pypi.org/simple/setuptools-scm/: [Errno 101] Network is unreachable -- Some packages may not be found!
    Couldn't find index page for 'setuptools_scm' (maybe misspelled?)
    Download error on https://pypi.org/simple/: [Errno 101] Network is unreachable -- Some packages may not be found!
    No local packages or working download links found for setuptools_scm
    Traceback (most recent call last):
```
Rather than clone the repo as suggested on the Tier 2 Setup Instructions page I thought i'd try:
`pip install --proxy=https://XXXX:8888 setuptools_scm`

Then I installed cocotb[bus] as above and it seems to have installed ok.
`Successfully installed cocotb-1.5.1 cocotb-bus-0.1.1`

**Comment by ktbarrett** [comment]
What you are seeing seems like a bug. It doesn't seem to be passing the proxy setting to the isolation (PEP517) backend. Who I was targeting when making that documentation are those people in closed areas with no internet access. But for those who are simply behind a proxy, this is a good solution, thanks! I'll add it to the Wiki.
